utils/lit/lit/formats/dxc_hash.py

```diff
@@ -0,0 +1,234 @@
+from __future__ import absolute_import
+import os
+import subprocess
+import platform
+import filecmp
+from glob import iglob
+
+import lit.Test
+import lit.TestRunner
+import lit.util
+import lit.ShUtil as ShUtil
+from .base import TestFormat
+
+kIsWindows = platform.system() == 'Windows'
+
+# Don't use close_fds on Windows.
+kUseCloseFDs = not kIsWindows
+
+class ShellEnvironment(object):
+
+    """Mutable shell environment containing things like CWD and env vars.
+
+    Environment variables are not implemented, but cwd tracking is.
+    """
+
+    def __init__(self, cwd, env):
+        self.cwd = cwd
+        self.env = env
+
+class DxcHashTest(TestFormat):
```

Comment:
Can we make the hash stability logic a standalone python script that we could run individually rather than buyring it inside a lit test format?

That would make it a lot easier to reproduce any individual hash stability test case failure.

Reply:
Sure. Where should the new hash_stability.py live?

Reply:
We've put scripts like this in utils/hct in the past. We've also used `CamelCase` naming for them.